src/Components/ArticleComments.jsx

```diff
@@ -1,31 +1,39 @@
 import { useEffect, useState } from "react";
 import { getCommentsbyArticleID } from "../api";
 import CommentCard from "./CommentCard";
+import PostComment from "./PostComment";
 
 const ArticleComments = ({ article_id }) => {
     const [comments, setComments] = useState([]); 
     const [isLoading, setIsLoading] = useState(true); 
+    const [newComment, setNewComment] = useState({})
 
     useEffect(() => {
         getCommentsbyArticleID(article_id)
         .then(({ data }) => {
+            data.sort((a, b) => Date.parse(b.created_at) - Date.parse(a.created_at))
             setComments(data); 
             setIsLoading(false); 
         })
-    }, [])
+    }, [article_id, newComment])
 
 const commentFormat = comments.map((comment) => {
-        return <CommentCard comment={comment}/>
+        return <CommentCard key={comment.comment_id}comment={comment}/>
     })
```

Comment:
You're getting the "every child must a `key`" warning from react, because when you are optimistically adding the new comment to the array, it doesn't have a `comment_id`. This is unavoidable if you choose to optimistically render it, so you will have to provide at least a custom id for a new comment just to suppress the warning!